autosklearn/pipeline/components/base.py

```diff
@@ -145,12 +147,21 @@ def __str__(self):
 
 class IterativeComponent(AutoSklearnComponent):
     def fit(self, X, y, sample_weight=None):
-        self.iterative_fit(X, y, n_iter=2, refit=True)
-        iteration = 2
-        while not self.configuration_fully_fitted():
-            n_iter = int(2 ** iteration / 2)
-            self.iterative_fit(X, y, n_iter=n_iter, refit=False)
-            iteration += 1
+        with warnings.catch_warnings():
```

Comment:
Is this to not have this warning shown in the unit tests?

Reply:
Well more in general but yes it also is for the unittests, I'm not sure where these warnings get swallowed in general, perhaps to the log file. We are aware that iterative algorithms will usually complain about convergence due to how we do things so I thought it would be best to consume them.

Thought's on alternatives:
1. Leave this specific warning to be omitted?
2. Other?

Reply:
Yes, they are redirected to the logfile in this function: https://github.com/automl/auto-sklearn/blob/62a66731b13be2b1c4002c232d89e4d3b211d268/autosklearn/evaluation/abstract_evaluator.py#L153

For that reason I'd prefer to let them reach the logfile instead. Would it be possible to swallow these in the unit tests? I think we can also remove quite a few warnings by updating `autosklearn.pipeline.components.base:IterativeComponent.fit` to be
```
    def fit(self, X, y, sample_weight=None):
        self.iterative_fit(X, y, n_iter=self.get_max_iter(), refit=True)
        return self
```
which as a nice side-effect could make Auto-sklearn as a whole faster (because stuff like GradientBoosting doesn't have to do initialization on calls to fit).

Reply:
We actually get different results if we do it this way, the assertion errors for explicit values change (minorly).